### DISCUSSION TITLE
Adonis v5 oauth

### DISCUSSION THREAD

**Discussion by Disbot-lab** [adonisjs/core#1444] [Help]
Hello, why do the fields `password` and `remember_me_token` have to be present to use auth? I only use oauth, is it possible to do differently?
[https://preview.adonisjs.com/guides/auth/setup#run-migrations](https://preview.adonisjs.com/guides/auth/setup#run-migrations)

**Comment by thetutlage** [comment]
The Auth library is not an implementation of `OAuth` and can you explain how login works in your case?

**Reply by Disbot-lab** [reply]
I would like to use auth with the `.loginViaId()` function but an error tells me that I need the `password` and `remember_me_token` fields when I don't need them because I already have all the necessary information thanks to the discord oauth.
In my login page I don't even have a form

**Reply by thetutlage** [reply]
Got it. I believe, the error you are talking about is the typescript compiler error and not a runtime one. We can make these properties optional in types.

Can you please open an issue regarding the same in `@adonisjs/auth` repo?

**Reply by Disbot-lab** [reply]
It's done https://github.com/adonisjs/auth/issues/147
